client-js/main/client/firebase-subscription-service.js

```diff
@@ -125,21 +125,40 @@ export class FirebaseSubscriptionService {
    * @private
    */
   _keepUpSubscriptions() {
-    this._subscriptions.forEach(subscription => {
-      const spineSubscription = subscription.internal();
-      if (subscription.closed) {
-        this._endpoint.cancelSubscription(spineSubscription).then(() => {
-          this._removeSubscription(subscription);
-        });
-      } else {
-        this._endpoint.keepUpSubscription(spineSubscription).then(response => {
-          const responseStatus = response.status;
-          const responseStatusProto = ObjectToProto.convert(responseStatus, Status.typeUrl());
-          if (responseStatusProto.getStatusCase() !== Status.StatusCase.OK) {
-            this._removeSubscription(subscription)
-          }
-        });
-      }
+    this._cancelClosed()
+    this._keepUpOrGiveUp();
+  }
+
+  _cancelClosed() {
```

Comment:
I would document this one and the one below. 

The first one is pretty mysterious by name. And the second requires some clarifications on that "Or".